src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/TestStreamedResponseHtml.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Symfony\Bundle\WebProfilerBundle\Tests\EventListener;
+
+use Symfony\Component\HttpFoundation\HtmlResponseInterface;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+/**
+ * @internal
+ *
+ * @author Bartłomiej Krukowski <bartlomiej@krukowski.me>
+ */
+class TestStreamedResponseHtml extends StreamedResponse implements HtmlResponseInterface
+{
+    private $extraHtml = '';
+
+    public function appendToBody($html)
+    {
+        $this->extraHtml .= $html;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendContent()
+    {
+        ob_start();
+        parent::sendContent();
+        $content = ob_get_contents();
+        ob_end_clean();
```

Comment:
doing an implementation based on a bugger totally defeats the purpose of using a streamed response, as it is not streamed anymore. So this does not show that you can solve your issue (using your TestStreamedResponseHtml ends up being equivalent to using a normal Response regarding the sending of the response)